src/common/src/hash/key.rs

```diff
@@ -571,7 +605,8 @@ impl<const N: usize> HashKeyDeserializer for FixedSizeKeyDeserializer<N> {
 pub struct SerializedKeySerializer {
     buffer: Vec<u8>,
     hash_code: u64,
-    null_bitmap: FixedBitSet,
+    null_bitmap: NullBitmap,
+    null_bitmap_idx: usize,
```

Comment:
1. the buffer is a vector which must be on the heap instead of stack, so the optimization for null_bitmap not very effiecent.
2. the u8 might could violate the Memory Alignment? I am not very sure about it 🤔 can we use Set32 and Set64 to have a try?

Reply:
> the buffer is a vector which must be on the heap instead of stack, so the optimization for null_bitmap not very effiecent.

To avoid this cost we have to first change `SerializedKey`'s representation since it currently uses `Vec<u8>`.
```rs

/// Designed for hash keys which can't be represented by [`FixedSizeKey`].
///
/// See [`crate::hash::calc_hash_key_kind`]
#[derive(Clone, Debug)]
pub struct SerializedKey {
    // Key encoding.
    key: Vec<u8>, // Key here is using `Vec`.
    hash_code: u64,
    null_bitmap: FixedBitSet,
}

```

1. Imo one approach could be to allocate a chunk of memory, represent the `key` as just offsets to this chunk of memory.
This avoids cost of calling `allocate` so often. But memory cost will still be the same I guess.

2. Another approach could be to represent the `key` in a series of fixed sizes in [N; u64] blocks, where N is a pre-determined fixed constant. Then it will reside on the stack.


> the u8 might could violate the Memory Alignment? I am not very sure about it 🤔 can we use Set32 and Set64 to have a try?
Agree. Preliminary results for Set64 look promising. Btw this will also result in breaking change. It sets a hard limit on the number of group keys we can have to 64. I think that is adequate for most usecases. Postgres itself has a default of 32 for partition keys.